default.nix

```diff
@@ -0,0 +1,247 @@
+  dependencyContent = pages: lib.lists.unique (
+    (map (dep: dep.content) standardDependencies) ++
+    (builtins.concatMap (page: dependencyContentRecur page.dependencies) pages)
+  );
+
+  renderBooklet = booklet:
+    let
+      addPage = content: pageData: {
+        content = content.content + lib.optionalString (pageData ? sourceHeader) ''
+          %****************************************************************
+          %*
+          %* ${pageData.sourceHeader}
+          %*
+          %****************************************************************
+        '' + ''
+          %%Page: ${toString content.nextPgIdx} ${toString content.nextPgIdx}
+          ${lib.optionalString (pageData ? isLandscape) "%%PageOrientation: Landscape\n"}%%BeginPageSetup
+          /pgsave save def
+          %%EndPageSetup
+        '' + (
+          if pageData ? drawPageContent
+          then
+            if pageData ? isLandscape
+            then "landscapePage begin ${toString content.nextCtIdx} drawPageContent\n"
+            else "portraitPage begin ${toString content.nextCtIdx} drawPageContent\n"
+          else
+            if pageData ? isLandscape
+            then "90 rotate\n"
+            else ""
+        ) + ''
+          ${pageData.content}
+          ${lib.optionalString (pageData ? drawPageContent) "end\n"}pgsave restore
+          showpage
+        '';
+        nextCtIdx = content.nextCtIdx + (if pageData ? drawPageContent then 1 else 0);
+        nextPgIdx = content.nextPgIdx + 1;
+      };
+      initialContent = {
+        content = ''
+          %!PS-Adobe-3.0
+          %%Orientation: Portrait
+          %%Pages: ${toString (builtins.length booklet.pages)}
+          %%EndComments
+          %%BeginSetup
+          ${toString (dependencyContent (booklet.pages))}%%EndSetup
+
+          %************************************************************************
+          %************************************************************************
+          %*
+          %* Section Three: Page Rendering
+          %*
+          %************************************************************************
+          %************************************************************************
+
+        '';
+        nextPgIdx = 1;
+        nextCtIdx = 1;
+      };
+      finalContent = builtins.foldl' addPage initialContent booklet.pages;
+    in
+    pkgs.writeTextFile {
+      name = booklet.name;
+      text = finalContent.content + ''
+        %%EOF
+      '';
+    };
+in
+stdenv.mkDerivation {
+  name = "codex32${shortId}";
+
+  buildInputs = [ ghostscript ];
+
+  phases = [ "buildPhase" ];
+  buildPhase = ''
+    set -e
+
+    ghostscriptTest() {
+      echo "Ghostscript testing $1"
+      local output;
+      if ! output="$(gs -dNOPAUSE -dNODISPLAY "$1" < /dev/null 2>&1)"; then
+        echo "Failed to run ghostscript on $1"
+        echo "$output"
+        exit 1
+      fi
+    }
+  '' + lib.optionalString doPageChecks toString (
```

Comment:
Should this all be in a standard `checkPhase` gated by a standard `doCheck` flag?

Reply:
Ah, yes, it should be.